autogpt_platform/frontend/src/app/(platform)/build/components/FlowEditor/nodes/fields/CredentialField/models/APIKeyCredentialModal/APIKeyCredentialModal.tsx

```diff
@@ -0,0 +1,119 @@
+import { Input } from "@/components/atoms/Input/Input";
+import { Button } from "@/components/atoms/Button/Button";
+import { Dialog } from "@/components/molecules/Dialog/Dialog";
+import {
+  Form,
+  FormDescription,
+  FormField,
+} from "@/components/__legacy__/ui/form";
+import { BlockIOCredentialsSubSchema } from "@/lib/autogpt-server-api/types"; // we need to find a way to replace it with autogenerated types
+import { useAPIKeyCredentialsModal } from "./useAPIKeyCredentialsModal";
+import { toDisplayName } from "../../helpers";
+
+type Props = {
+  schema: BlockIOCredentialsSubSchema;
+  open: boolean;
+  onClose: () => void;
+  onSuccess: (credentialId: string) => void;
+};
+
+export function APIKeyCredentialsModal({
```

Comment:
We already have this one:
https://github.com/Significant-Gravitas/AutoGPT/blob/master/autogpt_platform/frontend/src/app/(platform)/library/agents/%5Bid%5D/components/AgentRunsView/components/APIKeyCredentialsModal/APIKeyCredentialsModal.tsx

Would be good we keep 1 modal for adding API keys, we can place it under `src/components/contextual` and re-use everywhere 💅🏽